src/pyhf/optimize/mixins.py

```diff
@@ -0,0 +1,137 @@
+"""Helper Classes for use of automatic differentiation."""
+from .. import get_backend, exceptions
+from .common import shim
+
+import logging
+
+log = logging.getLogger(__name__)
+
+
+class OptimizerMixin(object):
+    """Mixin Class to build optimizers."""
+
+    def __init__(self, **kwargs):
+        """
+        Create an optimizer.
+
+        Args:
+            maxiter (`int`): maximum number of iterations. Default is 100000.
+            verbose (`bool`): print verbose output during minimization. Default is off.
```

Comment:
As these are bools we should reflect that in the docs usage.

```suggestion
            verbose (`bool`): print verbose output during minimization. Default is ``False``.
```

Reply:
verbose changed to `int` and reworded again.